pkg/rancher-desktop/pages/Containers.vue

```diff
@@ -331,25 +331,27 @@ export default {
       return { content: sha };
     },
     getUniquePorts(ports) {
-      const keys = Object.keys(ports);
+      const displayMap = [];
 
-      const uniquePortMap = keys.map((key) => {
-        const values = ports[key];
-        const hostPorts = values.map(value => value.HostPort);
-        const uniqueHostPorts = [...new Set(hostPorts)];
+      Object.entries(ports).forEach((entry) => {
+        const [key, values] = entry;
 
-        return { [key]: uniqueHostPorts };
-      });
-
-      const displayMap = uniquePortMap.map((element) => {
-        const key = Object.keys(element)[0];
-        const values = element[key];
-        const port = key.split('/')[0];
-
-        return values.map(value => `${ value }:${ port }`);
+        if (!values) {
+          return;
+        }
+        try {
+          const hostPorts = (values || []).map(value => value.HostPort);
+          const port = key.split('/')[0];
+
+          (new Set(hostPorts)).forEach((hostPort) => {
```

Comment:
More trivial style things that don't matter in the grand scheme…
```suggestion
          for (const hostPort of (new Set(hostPorts))) {
```

Reply:
No need to parenthesize `new Set...` there either.